pulp_smash/tests/rpm/api_v2/test_schedule_consumer.py

```diff
@@ -0,0 +1,163 @@
+    """
+
+    def setUp(self):
+        """Install required packages and start the ``goferd`` service.
+
+        This is done because ``goferd`` must be running in order to communicate
+        with ``pulp-consumer-client``.
+        """
+        self.cfg = config.get_config()
+        if selectors.bug_is_untestable(2680, self.cfg.version):
+            self.skipTest('https://pulp.plan.io/issues/2680')
+        self.pkg_not_installed = []
+        cli_client = cli.Client(self.cfg)
+        for package in REQUIRED_PACKAGES:
+            try:
+                cli_client.run(('rpm', '-q', package))
+            except exceptions.CalledProcessError:
+                self.pkg_not_installed.append(package)
+        self.pkg_mgr = cli.PackageManager(self.cfg)
+        if self.pkg_not_installed:
+            self.pkg_mgr.install(self.pkg_not_installed)
+        system = self.cfg.get_systems('shell')[0]
+        self.svc_mgr = cli.ServiceManager(self.cfg, pulp_system=system)
+        self.svc_mgr.start(REQUIRED_SERVICES)
+        self.sudo = () if utils.is_root(self.cfg) else ('sudo',)
+
+    def tearDown(self):
+        """Remove registered consumer and stop ``goferd`` service.
+
+        Un-install previous installed packages.
+        """
+        self.svc_mgr.stop(REQUIRED_SERVICES)
+        cli.Client(self.cfg).run(self.sudo + (
+            'pulp-consumer', '-u', self.cfg.pulp_auth[0], '-p',
+            self.cfg.pulp_auth[1], 'unregister', '--force'
+        ))
+        if self.pkg_not_installed:
+            self.pkg_mgr.uninstall(self.pkg_not_installed)
+
+    def test_all(self):
+        """Verify scheduled RPM installation and un-installation work.
+
+        Do the following:
+        1. Create a consumer, and register it.
+        2. Create, sync and publish a repo.
+        3. Bind the consumer to the created repo.
+        4. Schedule the installation of RPM package.
+        5. Schedule the un-installation of the installed package.
+        """
+        cli_client = cli.Client(self.cfg)
+
+        # Create and register a consumer.
+        consumer_id = utils.uuid4()
+        cli_client.run(self.sudo + (
+            'pulp-consumer', '-u', self.cfg.pulp_auth[0], '-p',
+            self.cfg.pulp_auth[1], 'register', '--consumer-id', consumer_id
+        ))
+
+        # Create, sync and publish a repository.
+        client = api.Client(self.cfg, api.json_handler)
+        body = gen_repo()
+        body['importer_config']['feed'] = RPM_UNSIGNED_FEED_URL
+        body['distributors'] = [gen_distributor()]
+        repo = client.post(REPOSITORY_PATH, body)
+        self.addCleanup(client.delete, repo['_href'])
+        utils.sync_repo(self.cfg, repo)
+        repo = client.get(repo['_href'], params={'details': True})
+        utils.publish_repo(self.cfg, repo)
+
+        # Bind the consumer.
+        client.post(urljoin(CONSUMERS_PATH, consumer_id + '/bindings/'), {
```

Comment:
The repeated references to `CONSUMERS_PATH` and `consumer_id` make me think that a `consumer_path` variable should be created.